tests/everflow/test_everflow_testbed.py

```diff
@@ -841,6 +841,121 @@ def background_traffic(run_count=None):
+
+        def configure_mirror_session_with_queue(mirror_session, queue_num):
+            if mirror_session["session_name"]:
+                remove_command = "config mirror_session remove {}".format(mirror_session["session_name"])
+                for duthost in duthost_set:
+                    duthost.command(remove_command)
+                add_command = "config mirror_session add {} {} {} {} {} {} {}" \
+                    .format(mirror_session["session_name"],
+                            mirror_session["session_src_ip"],
+                            mirror_session["session_dst_ip"],
+                            mirror_session["session_dscp"],
+                            mirror_session["session_ttl"],
+                            mirror_session["session_gre"],
+                            queue_num)
+                for duthost in duthost_set:
+                    duthost.command(add_command)
+            else:
+                pytest.skip("Mirror session info is empty, can't proceed further!")
+
+        queue = str(random.randint(1, 7))
+        # Apply mirror session config with a different queue value other than default '0'
+        configure_mirror_session_with_queue(setup_mirror_session, queue)
+
+        # Add a route to the mirror session destination IP
+        tx_port = setup_info[dest_port_type]["dest_port"][0]
+        peer_ip = everflow_utils.get_neighbor_info(remote_dut, tx_port, tbinfo)
+        everflow_utils.add_route(remote_dut, setup_mirror_session["session_prefixes"][0], peer_ip,
+                                 setup_info[dest_port_type]["remote_namespace"])
+
+        time.sleep(15)
+
+        # Verify that mirrored traffic is sent along the route we installed
+        rx_port_ptf_id = setup_info[dest_port_type]["src_port_ptf_id"]
+        tx_port_ptf_id = setup_info[dest_port_type]["dest_port_ptf_id"][0]
+        src_port = setup_info[dest_port_type]["src_port"]
+        # Clear queue counters for the port asic instance
+        asic_ns = everflow_dut.get_port_asic_instance(src_port).get_asic_namespace()
+        if asic_ns is not None:
+            asic_id = everflow_dut.get_asic_id_from_namespace(asic_ns)
+            recircle_port = "Ethernet-Rec{}".format(asic_id)
+        else:
+            recircle_port = "Ethernet-Rec"
+
+        try:
+            everflow_utils.verify_mirror_packets_on_recircle_port(
+                self,
+                ptfadapter,
+                setup_info,
+                setup_mirror_session,
+                everflow_dut,
+                rx_port_ptf_id,
+                [tx_port_ptf_id],
+                dest_port_type,
+                queue,
+                asic_ns,
+                recircle_port
+            )
+        finally:
+            remote_dut.shell(remote_dut.get_vtysh_cmd_for_namespace(
+                "vtysh -c \"configure terminal\" -c \"ip nht resolve-via-default\"",
+                setup_info[dest_port_type]["remote_namespace"]))
```

Comment:
Do we need to remove the routes or acls the test setup too?

Reply:
Yes, we do.  All those fixtures are invoked during test teardown. For example, '_add_dest_routes_ ', '_dest_port_type_ ', '_setup_acl_table_ ', '_setup_mirror_session_ ', '_setup_arp_responder_ ' and '_setup_info_'. Attaching the test run log for reference.
[test_everflow_rec_queue.txt](https://github.com/user-attachments/files/20711930/test_everflow_rec_queue.txt)